ray-operator/controllers/ray/rayjob_controller_suspended_test.go

```diff
@@ -17,242 +17,177 @@ package ray
-						MinReplicas:    pointer.Int32(0),
-						MaxReplicas:    pointer.Int32(10000),
-						GroupName:      "small-group",
-						RayStartParams: map[string]string{},
-						Template: corev1.PodTemplateSpec{
-							Spec: corev1.PodSpec{
-								Containers: []corev1.Container{
-									{
-										Name:  "ray-worker",
-										Image: "rayproject/ray:2.8.0",
-									},
-								},
-							},
-						},
-					},
-				},
-			},
-		},
-	}
-
+var _ = Context("RayJob with suspend operation", func() {
 	Describe("When creating a rayjob with suspend == true", func() {
+		ctx := context.Background()
+		namespace := "default"
+		rayCluster := &rayv1.RayCluster{}
+		rayJob := rayJobTemplate("rayjob-suspend", namespace)
+		rayJob.Spec.Suspend = true
+
 		It("should create a rayjob object", func() {
-			err := k8sClient.Create(ctx, mySuspendedRayJob)
+			err := k8sClient.Create(ctx, rayJob)
 			Expect(err).NotTo(HaveOccurred(), "failed to create test RayJob resource")
 		})
 
-		It("should see a rayjob object", func() {
-			Eventually(
-				getResourceFunc(ctx, client.ObjectKey{Name: mySuspendedRayJob.Name, Namespace: "default"}, mySuspendedRayJob),
-				time.Second*3, time.Millisecond*500).Should(BeNil(), "My myRayJob  = %v", mySuspendedRayJob.Name)
-		})
-
 		It("should have deployment status suspended", func() {
 			Eventually(
-				getRayJobDeploymentStatus(ctx, mySuspendedRayJob),
+				getRayJobDeploymentStatus(ctx, rayJob),
 				time.Second*5, time.Millisecond*500).Should(Equal(rayv1.JobDeploymentStatusSuspended))
 		})
 
 		It("should NOT create a raycluster object", func() {
 			Consistently(
-				getRayClusterNameForRayJob(ctx, mySuspendedRayJob),
+				getRayClusterNameForRayJob(ctx, rayJob),
 				time.Second*3, time.Millisecond*500).Should(BeEmpty())
 		})
 
 		It("should unsuspend a rayjob object", func() {
-			mySuspendedRayJob.Spec.Suspend = false
-			err := k8sClient.Update(ctx, mySuspendedRayJob)
-			Expect(err).NotTo(HaveOccurred(), "failed to update test RayJob resource")
+			err := updateRayJobSuspendField(ctx, rayJob, false)
+			Expect(err).NotTo(HaveOccurred(), "failed to update RayJob")
 		})
 
 		It("should create a raycluster object", func() {
 			// Ray Cluster name can be present on RayJob's CRD
 			Eventually(
-				getRayClusterNameForRayJob(ctx, mySuspendedRayJob),
+				getRayClusterNameForRayJob(ctx, rayJob),
 				time.Second*15, time.Millisecond*500).Should(Not(BeEmpty()))
 			// The actual cluster instance and underlying resources SHOULD be created when suspend == false
 			Eventually(
 				// k8sClient client does not throw error if cluster IS found
-				getResourceFunc(ctx, client.ObjectKey{Name: mySuspendedRayJob.Status.RayClusterName, Namespace: "default"}, mySuspendedRayCluster),
+				getResourceFunc(ctx, client.ObjectKey{Name: rayJob.Status.RayClusterName, Namespace: namespace}, rayCluster),
 				time.Second*3, time.Millisecond*500).Should(BeNil())
 		})
 
-		It("should create 3 workers", func() {
```

Comment:
I removed `should create 3 workers` and `should create a head pod resource` because `updateHeadPodToRunningAndReady` and `updateWorkerPodsToRunningAndReady` functions now can automatically check the number of workers and the head pod.